piwik/piwik
# Update Dashboard when widget list is reloaded

### Description:

Came up with a fix for the issue with the `updateReportingMenu` event not updating the Manage Dashboard dropdown's collection of available widgets. I had to create a second event to listen to for once the store had been updated. This is related to the previous PR: https://github.com/matomo-org/matomo/pull/22671

Fixes: #12265 

Easiest test steps:
- Create a new custom report
- Without refreshing page, navigate to the dashboard and confirm that the new report is in the list of widgets for the Manage Dashboard dropdown
- Delete the custom report
- Without refreshing page, navigate to the dashboard and confirm that the deleted report is no longer in the widget list

### Review
